src/tests/ftest/util/telemetry_utils.py

```
@@ -960,3 +1055,251 @@ def verify_metric_value(self, metrics_data, min_value=None, max_value=None):
+                    {<metric>: {<label>: <min_value>}} or
+                    {<metric>: {<label>: [<min_value>]}} or
+                    {<metric>: {<label>: [<min_value>, <max_value>]}}
+
+        Returns:
+            bool: True if all metric values are within the ranges specified; False otherwise
+        """
+        status = self._set_display(ranges)
+        columns = ['metric'] + self._display['labels'] + ['value', 'check']
+        format_str = '  '.join([f"%-{self._display['widths'][name]}s" for name in columns])
+
+        log.info('-' * 80)
+        log.info('Telemetry Metric Verification')
+        log.info(format_str, *[name.title() for name in columns])
+        log.info(format_str, *['-' * self._display['widths'][name] for name in columns])
+        for metric in sorted(self._display['data']):
+            for value, labels in self._display['data'][metric].items():
+                log.info(
+                    format_str, metric, *self._label_values(labels), value,
+                    *self._label_values(labels, ['check']))
+        return status
+
+    def _get_metrics(self, log, names, hosts, dmg):
+        """Obtain the specified metric information for each host.
+
+        Args:
+            log (logger): logger for the messages produced by this method
+            names (str): Comma-separated list of metric names to query.
+            hosts (NodeSet): set of servers from which to collect the telemetry metrics
+            dmg (DmgCommand): the DmgCommand object configured to communicate with the servers
+
+        Returns:
+            dict: a dictionary of host keys linked to metric data for each metric name specified
+        """
+        info = {}
+        log.info('Querying telemetry metric %s from %s', names, hosts)
+        for host in hosts:
+            data = dmg.telemetry_metrics_query(host=host, metrics=names)
+            info[host] = {}
+            if 'response' in data:
```

And if it's not? This will silently skip over bad output